src/telem/lib/input/mekanism/DigitalMinerInputAdapter.lua

```diff
@@ -0,0 +1,67 @@
+local o = require 'telem.lib.ObjectModel'
+local t = require 'telem.lib.util'
+
+local InputAdapter      = require 'telem.lib.InputAdapter'
+local Metric            = require 'telem.lib.Metric'
+local MetricCollection  = require 'telem.lib.MetricCollection'
+
+local DigitalMinerInputAdapter = o.class(InputAdapter)
+DigitalMinerInputAdapter.type = 'DigitalMinerInputAdapter'
+
+function DigitalMinerInputAdapter:constructor (peripheralName, categories)
+    self:super('constructor')
+
+    -- TODO this will be a configurable feature later
+    self.prefix = 'mekdigitalminer:'
+
+    -- TODO make these constants
+    local allCategories = {
+        'basic',
+    }
+
+    if not categories then
+        self.categories = { 'basic' }
+    elseif categories == '*' then
+        self.categories = allCategories
+    else
+        self.categories = categories
+    end
+
+    -- boot components
+    self:setBoot(function ()
+        self.components = {}
+
+        self:addComponentByPeripheralID(peripheralName)
+    end)()
+end
+
+function DigitalMinerInputAdapter:read ()
+    self:boot()
+
+    local source, miner = next(self.components)
+
+    local metrics = MetricCollection()
+
+    local loaded = {}
+
+    for _,v in ipairs(self.categories) do
+        -- skip, already loaded
+        if loaded[v] then
+            -- do nothing
+
+        -- Literally all we have lmao
+        elseif v == 'basic' then
```

Comment:
Categories should be expanded:

Basic:
- energy_filled_percentage
- energy_usage
- get_to_mine
- is_running

Energy:
- energy